lib/models/api_calls.dart

```diff
@@ -80,20 +83,20 @@ class ApiCalls {
   ///
   /// This method returns a [DetailedDevice] object.
   dynamic getDetailedDevice(String authToken, String deviceId) async {
+    var response;
     authToken = 'Bearer $authToken';
-    try {
-      final response = await _client.get(DETAILED_DEVICE_URL + deviceId,
-          headers: {HttpHeaders.authorizationHeader: authToken});
-      // If conection is succesful return a DetailedDevice object by parsing the
-      // response body
-      if (response.statusCode == 200) {
-        return DetailedDevice.fromMap(json.decode(response.body));
-      } else {
-        // If that call was not successful, throw an error.
-        throw Exception(response.statusCode);
-      }
-    } finally {
-      _client.close();
+    _client == null
```

Comment:
Same, check if you can cache the client to use, to attempt to simplify the code.

Reply:
http is a prefix to a library so I cant cache the client.

Reply:
does it allow to cache functions? So you would do something like var getFunction = _client == null ? http.get ? _client.get;
getFunction.apply(url, headers)